src/MPQEditor/MPQEditor.ts

```diff
@@ -0,0 +1,530 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as cp from "child_process";
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+
+import { Balloon } from "../Utils/Balloon";
+import { Logger } from "../Utils/Logger";
+import { getNonce } from "../Utils/external/Nonce";
+import { getUri } from "../Utils/external/Uri";
+import { MPQData } from "./MPQData";
+
+export class MPQEditorProvider implements vscode.CustomTextEditorProvider {
+  public static readonly viewType = "one.editor.mpq";
+  public static readonly fileExtension = ".mpq.json";
+
+  private _disposables: vscode.Disposable[] = [];
+  private _mpqDataMap: any = {};
+
+  /**
+   * register commands of MPQEditorProvider
+   */
+  public static register(context: vscode.ExtensionContext): void {
+    const provider = new MPQEditorProvider(context);
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(
+        MPQEditorProvider.viewType,
+        provider,
+        {
+          webviewOptions: { retainContextWhenHidden: true },
+        }
+      ),
+      // Add command registration here
+      vscode.commands.registerCommand(
+        "one.editor.mpq.showFromDefaultExplorer",
+        (uri) => {
+          MPQEditorProvider.createMPQJson(uri);
+        }
+      ),
+    ];
+
+    registrations.forEach((disposable) =>
+      context.subscriptions.push(disposable)
+    );
+  }
+
+  /**
+   * find unoccupied mpq file-name
+   */
+  public static findMPQName(baseMPQName: string, dirPath: string): string {
+    const maxIters = 5;
+    for (let i = 0; i < maxIters; i++) {
+      const mpqPath: string = path.join(
+        dirPath,
+        baseMPQName + MPQEditorProvider.fileExtension
+      );
+      if (!fs.existsSync(mpqPath)) {
+        return baseMPQName + MPQEditorProvider.fileExtension;
+      }
+      baseMPQName = baseMPQName + "(1)";
+    }
+    return "";
+  }
```

Comment:
This function will not work as expected when the trial repeats more than 5 times.
It will return the "", which is not expected.

On top of that, I think it's natural to have increasing numbers (1)->(2)->...
Could you revise this function?

Reply:
Yes. I can. Thank you.

Reply:
@dayo09 
I've implemented (and updated tests and draft description) simple version of it , in the form of
```
for (let i =0; ; i++)
```
do i need to constrain somehow index (by 10000 e.g.) ?
Or it was assumed to some form `glob` query , e.g.
```
1. find the longest filename with the pattern *(*).mpq.json
2. extract number from round brackets
3. return number + 1 as possible answer
```
?